src/modules/services/dynamodb-local.nix

```diff
@@ -0,0 +1,40 @@
+{ pkgs, lib, config, ... }:
+
+let
+  cfg = config.services.dynamodb-local;
+  types = lib.types;
+  baseDir = config.env.DEVENV_STATE + "/dynamodb-local";
+  startScript = pkgs.writeShellScript "start-dynamodb-local" ''
+    set -euo pipefail
+
+    if [[ ! -d "${baseDir}" ]]; then
+      mkdir -p "${baseDir}"
+    fi
+
+    cd "${baseDir}"
+
+    ${config.services.dynamodb-local.package}/bin/dynamodb-local -port ${toString cfg.port} -dbPath ${baseDir} -disableTelemetry
+  '';
+in
+{
+  options.services.dynamodb-local = {
+    enable = lib.mkEnableOption "DynamoDB Local";
+
+    package = lib.mkOption {
+      type = types.package;
+      description = "Which package of DynamoDB to use.";
+      default = pkgs.dynamodb-local;
+      defaultText = lib.literalExpression "pkgs.dynamodb-local";
+    };
+
+    port = lib.mkOption {
+      type = types.port;
+      description = "Listen address for the Dynamodb-local.";
+      default = 8000;
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
```

Comment:
Consider adding `process-compose` specific configuration as well:
https://github.com/cachix/devenv/blob/ff37923d4ba9e220b9826594e83fc29e1a4f4338/src/modules/services/postgres.nix#L298-L313

Reply:
what would be the readiness command? kinda something like curl? as here is nothing special 🤔

Reply:
https://github.com/cachix/devenv/blob/ff37923d4ba9e220b9826594e83fc29e1a4f4338/src/modules/services/opensearch.nix#L158
Yeah, something like this would be great. With a lower initial_delay_second, as I guess this starts relatively fast.

Reply:
https://github.com/cachix/devenv/blob/ff37923d4ba9e220b9826594e83fc29e1a4f4338/src/modules/services/memcached.nix#L57-L59
Or just checking with netcat whether the port is bound. Even faster